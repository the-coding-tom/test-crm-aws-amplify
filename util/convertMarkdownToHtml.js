export const convertMarkdownToHtml = text => {
  var showdown = require('showdown')
  var converter = new showdown.Converter()
  const eventDescription = text.replace(/([ ]*\n)/g, `<br/>\n`)
  const finalDescription = eventDescription.replace(
    /(<br\/>\n<br\/>\n<br\/>)/g,
    `<br/><br/>`
  )
  return converter.makeHtml(finalDescription)
}
