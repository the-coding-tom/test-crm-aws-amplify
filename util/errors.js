export const displayError = function(e) {
  if (e.response) {
    processError(e.response.data.message)

    if (e.response.data.errors) {
      _.each(e.response.data.errors, (o, i) => {
        return _.each(o, (v, k) => processError(v))
      })
    }

    return
  }

  processError(e.message)
}

const processError = function(message) {
  console.log('------------------------------------')
  console.log(message)
  console.log('------------------------------------')
}
