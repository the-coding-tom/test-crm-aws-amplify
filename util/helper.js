const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

const parseRoomdate = (data, vm) => {
  const roomAvaila = data.split(',')
  const roomAvailability = []
  roomAvaila.forEach(data => {
    let weekday = []
    let timeof = []
    let lookup = {
      su: 0,
      mo: 1,
      tu: 2,
      we: 3,
      th: 4,
      fr: 5,
      sa: 6
    }
    const timeData = data.split(' ')
    if (timeData.length === 2) {
      weekday.push(timeData[0])
      timeof.push(timeData[1])
    }
    if (timeData.length === 3) {
      weekday.push(timeData[1])
      timeof.push(timeData[2])
    }
    const splitWeekday = weekday[0].split('-')
    const splitTimeOf = timeof[0].split('-')
    const AStruc = {
      from: '',
      to: '',
      weekdays: []
    }
    splitWeekday.forEach(week => {
      AStruc.weekdays.push(lookup[week])
    })
    AStruc.from = vm.$moment(splitTimeOf[0], 'HH').format('hh:mm')
    AStruc.to = vm.$moment(splitTimeOf[1], 'HH').format('hh:mm')
    roomAvailability.push(AStruc)
  })
  return roomAvailability
}

const undoParse = (data, vm) => {
  let str = ''
  const lookup = {
    0: 'su',
    1: 'mo',
    2: 'tu',
    3: 'we',
    4: 'th',
    5: 'fr',
    6: 'sa'
  }
  data.forEach(item => {
    let available = ''
    item.weekdays.forEach(day => {
      available += `${lookup[day]}-`
    })
    available = available.slice(0, -1)
    let weekdayfrom = item.from.charAt(0) + item.from.charAt(1)
    let weekdayto = item.to.charAt(0) + item.to.charAt(1)
    available += ` ${weekdayfrom}-${weekdayto}`
    str += `${available},`
  })
  str = str.slice(0, -1)
  return str
}

export default { randomColor, parseRoomdate, notify, parseRoomdate, undoParse }
