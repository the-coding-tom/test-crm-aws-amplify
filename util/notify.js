const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

export default notify
