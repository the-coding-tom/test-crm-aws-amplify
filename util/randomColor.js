const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

export default randomColor
