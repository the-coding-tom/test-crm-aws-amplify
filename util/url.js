export const getQueryParams = url => {
  const urlParam = require('url')

  const parsedUrl = urlParam.parse(url)

  return parsedUrl.search
}
