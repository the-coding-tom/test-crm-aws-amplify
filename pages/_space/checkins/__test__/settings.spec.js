import SettingsPage from '@/pages/_space/checkins/settings.vue'
import BootstrapVue from 'bootstrap-vue'
import BaseHeader from '~/components/argon-core/BaseHeader'

import VueRouter from 'vue-router'
import cloneDeep from 'lodash/cloneDeep'
import flushPromises from 'flush-promises'
import { createWrapper } from '@/test.util.js'
import lodash from 'lodash'

import allSettings from './resources/all-settings.json'
import currentSpace from '@/pages/_space/__test__/resources/current-space.json'

describe('Page: /:space/checkins/settings | Checkins Settings Index Page', () => {
  let NuxtStore, store, wrapper, $checkin, router

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)

    createRouter()

    $checkin = {
      saveSetting: jest.fn().mockResolvedValue(),
      deleteSetting: jest.fn().mockResolvedValue(),
      settings: jest.fn().mockResolvedValue(cloneDeep(allSettings))
    }

    global._ = lodash

    // mock reload
    const { location } = window
    delete window.location
    window.location = { ...location, reload: jest.fn() }
  })

  beforeEach(async () => {
    store = await NuxtStore.createStore()
    store.commit('space/setSpace', currentSpace)

    router = new VueRouter({
      routes: []
    })

    const asyncDataResponse = await SettingsPage.asyncData({ store, $checkin })

    wrapper = createWrapper(
      SettingsPage,
      store,
      router,
      {
        loading: false,
        fields: [
          {
            key: 'type',
            label: 'Membership Type'
          },
          {
            key: 'option',
            label: 'Number of Guests'
          },
          {
            key: 'action',
            label: ''
          }
        ],
        setting: {
          option: '1',
          type: ''
        },
        ...asyncDataResponse
      },
      {},
      vueInstance => {
        vueInstance.use(BootstrapVue)
        vueInstance.component(BaseHeader.name, BaseHeader)
        vueInstance.prototype.$checkin = $checkin
      }
    )
  })

  it('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('adds a setting', async () => {
    wrapper.find('#add-setting-modal').trigger('click')
    await wrapper.vm.$nextTick()

    const settingOptions = wrapper.findAll('select option')
    settingOptions.at(2).setSelected()
    await wrapper.vm.$nextTick()

    wrapper.find('input').setValue(20)
    await wrapper.vm.$nextTick()

    wrapper.find('form').trigger('submit.prevent')
    await flushPromises()

    expect($checkin.saveSetting).toHaveBeenCalled()
    expect(parseInt($checkin.saveSetting.mock.calls[0][1].option)).toBe(20)
    expect($checkin.saveSetting.mock.calls[0][1].type).toBe('1')
  })

  it('edits a setting', async () => {
    wrapper.find('button.btn.btn-primary.btn-sm').trigger('click')
    await flushPromises()

    const settingOptions = wrapper.findAll('select option')
    settingOptions.at(4).setSelected()
    await wrapper.vm.$nextTick()

    wrapper.find('input').setValue(60)
    await wrapper.vm.$nextTick()

    wrapper.find('form').trigger('submit.prevent')
    await flushPromises()

    expect($checkin.saveSetting).toHaveBeenCalled()
    expect(parseInt($checkin.saveSetting.mock.calls[1][1].option)).toBe(60)
    expect($checkin.saveSetting.mock.calls[1][1].type).toBe('3')
  })

  it('removes a setting', async () => {
    // mock confirm
    global.confirm = jest.fn().mockReturnValue(true)

    wrapper
      .find('button.btn.text-danger.btn-transparent.btn-sm')
      .trigger('click')

    expect($checkin.deleteSetting).toHaveBeenCalled()
    expect($checkin.deleteSetting.mock.calls[0][1]).toBe(allSettings.data[0].id)
  })
})
