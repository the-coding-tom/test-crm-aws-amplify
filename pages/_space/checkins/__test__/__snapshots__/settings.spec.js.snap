// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/checkins/settings | Checkins Settings Index Page renders properly 1`] = `
<div>
  <div class="header pb-6">
    <div class="container-fluid">
      <div class="header-body">
        <div class="d-flex justify-content-between align-items-center py-4">
          <h1 class="page-title"><span>Check-in</span> <b>•</b> Settings</h1> <button id="add-setting-modal" type="button" class="btn btn-primary">New Setting</button>
        </div>
      </div>
    </div>
  </div>
  <div class="container-fluid mt--6">
    <div class="row">
      <div class="col">
        <div class="card">
          <!---->
          <!---->
          <table role="table" aria-busy="false" aria-colcount="3" class="table b-table table-striped table-hover" id="__BVID__8">
            <!---->
            <!---->
            <thead role="rowgroup" class="">
              <!---->
              <tr role="row" class="">
                <th role="columnheader" scope="col" aria-colindex="1" class="">Membership Type</th>
                <th role="columnheader" scope="col" aria-colindex="2" class="">Number of Guests</th>
                <th role="columnheader" scope="col" aria-colindex="3" aria-label="Action" class=""></th>
              </tr>
            </thead>
            <tbody role="rowgroup">
              <!---->
              <tr role="row" class="">
                <td role="cell" aria-colindex="1" class=""><span>Founding Member</span></td>
                <td role="cell" aria-colindex="2" class="">6</td>
                <td role="cell" aria-colindex="3" class=""><button type="button" class="btn btn-primary btn-sm">Edit</button> <button type="button" class="btn text-danger btn-transparent btn-sm"><i class="ti ti-trash"></i> Remove</button></td>
              </tr>
              <tr role="row" class="">
                <td role="cell" aria-colindex="1" class=""><span>Fast Track</span></td>
                <td role="cell" aria-colindex="2" class="">3</td>
                <td role="cell" aria-colindex="3" class=""><button type="button" class="btn btn-primary btn-sm">Edit</button> <button type="button" class="btn text-danger btn-transparent btn-sm"><i class="ti ti-trash"></i> Remove</button></td>
              </tr>
              <tr role="row" class="">
                <td role="cell" aria-colindex="1" class=""><span>General Member</span></td>
                <td role="cell" aria-colindex="2" class="">3</td>
                <td role="cell" aria-colindex="3" class=""><button type="button" class="btn btn-primary btn-sm">Edit</button> <button type="button" class="btn text-danger btn-transparent btn-sm"><i class="ti ti-trash"></i> Remove</button></td>
              </tr>
              <tr role="row" class="">
                <td role="cell" aria-colindex="1" class=""><span>Early Invite</span></td>
                <td role="cell" aria-colindex="2" class="">3</td>
                <td role="cell" aria-colindex="3" class=""><button type="button" class="btn btn-primary btn-sm">Edit</button> <button type="button" class="btn text-danger btn-transparent btn-sm"><i class="ti ti-trash"></i> Remove</button></td>
              </tr>
              <!---->
              <!---->
            </tbody>
            <!---->
          </table>
          <!---->
          <!---->
        </div>
      </div>
    </div> <button type="button" class="btn mb-2 text-primary btn-transparent"><i class="fas fa-chevron-left"></i> Back</button>
  </div>
  <div id="modal-1___BV_modal_outer_" style="position: absolute; z-index: 1040;">
    <transition-stub enterclass="" leaveclass="" entertoclass="" leavetoclass="" enteractiveclass="" leaveactiveclass="">
      <div id="modal-1" role="dialog" aria-hidden="true" aria-labelledby="modal-1___BV_modal_title_" aria-describedby="modal-1___BV_modal_body_" class="modal fade" style="display: none;">
        <div class="modal-dialog modal-md">
          <!---->
          <div role="document" id="modal-1___BV_modal_content_" tabindex="-1" class="modal-content">
            <header id="modal-1___BV_modal_header_" class="modal-header">
              <h5 id="modal-1___BV_modal_title_" class="modal-title">Settings</h5><button type="button" aria-label="Close" class="close">×</button>
            </header>
            <div id="modal-1___BV_modal_body_" class="modal-body">
              <form class="">
                <fieldset class="form-group" id="__BVID__33">
                  <legend tabindex="-1" class="col-form-label pt-0" id="__BVID__33__BV_label_">Membership Type</legend>
                  <div tabindex="-1" role="group"><select required="required" aria-required="true" class="custom-select" id="__BVID__34">
                      <option value="" disabled="disabled">-- Choose a membership --</option>
                      <option value="0">Founding Member</option>
                      <option value="1">Fast Track</option>
                      <option value="2">Early Invite</option>
                      <option value="3">General Member</option>
                    </select>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
                <fieldset class="form-group" id="__BVID__35" aria-describedby="__BVID__35__BV_description_">
                  <legend tabindex="-1" class="col-form-label pt-0" id="__BVID__35__BV_label_">Number of Guests</legend>
                  <div tabindex="-1" role="group"><input type="number" placeholder="1" required="required" aria-required="true" class="form-control" id="__BVID__36">
                    <!---->
                    <!----><small tabindex="-1" class="form-text text-muted" id="__BVID__35__BV_description_">This is the number of guests the membership type can bring to the space as a general guest</small></div>
                </fieldset> <button type="submit" class="btn float-right btn-primary">Save</button>
              </form>
            </div>
            <!---->
          </div>
          <!---->
        </div>
      </div>
    </transition-stub>
    <transition-stub name="" css="true" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="fade" leaveactiveclass="fade">
      <!---->
    </transition-stub>
  </div>
</div>
`;
